src/ark/utils/data_utils.py

```diff
@@ -524,7 +524,7 @@ def stitch_images_by_shape(data_dir, stitched_dir, img_sub_folder=None, channels
             dir_name=os.path.join(data_dir, fovs[0], img_sub_folder),
             substrs=EXTENSION_TYPES["IMAGE"])
     else:
-        channel_imgs = io_utils.list_files(data_dir, substrs=fovs[0])
+        channel_imgs = io_utils.list_files(data_dir, substrs=fovs[0]+'_')
```

Comment:
Are you adding the `'_'` because there would be a conflict between `run_1` and `run_10`? If so, I've  also ran into that issue. Do you think it would be useful to implement a fix where numbers are treated properly?

Reply:
Yeah that's exactly why, but regarding the full FOV name. It's only coming up here because for segment/clustering mask directories there's no fov level folders, so I have to extract the "channel" names (e.g "whole_cell.tiff", "nuclear.tiff", etc.) and it was resulting in "0_whole_cell.tiff" for R1C1 if R1C10 exists.

tldr: I think it's fine for now, just something we have to be mindful of.